pkg/services/accesscontrol/accesscontrol.go

```diff
@@ -77,17 +80,42 @@ func (s *SearchOptions) Wildcards() []string {
 	return s.wildcards
 }
 
-func (s *SearchOptions) ResolveUserLogin(ctx context.Context, userSvc user.Service) error {
-	if s.UserLogin == "" {
+func (s *SearchOptions) ComputeUserID(ctx context.Context, userSvc user.Service) error {
+	if s.UserID > 0 {
 		return nil
 	}
+
+	// Resolve namespaceID -> userID
+	if s.NamespaceID != "" {
+		parts := strings.Split(s.NamespaceID, ":")
+		// Validate namespace ID format
+		if len(parts) != 2 {
+			return fmt.Errorf("invalid namespace ID: %s", s.NamespaceID)
+		}
+		// Validate namespace type is user or service account
+		if parts[0] != identity.NamespaceUser && parts[0] != identity.NamespaceServiceAccount {
+			return fmt.Errorf("invalid namespace type: %s", parts[0])
+		}
+		// Validate namespace ID is a number
+		id, err := strconv.ParseInt(parts[1], 10, 64)
+		if err != nil {
+			return fmt.Errorf("invalid namespace ID: %s", s.NamespaceID)
+		}
+		s.UserID = id
```

Comment:
We should probably check that the user with that ID really is a user or a service account, but dependency injection fails if I inject the service accounts service.